src/Microsoft.AspNet.FileProviders.Embedded/EmbeddedFileProvider.cs

```diff
@@ -100,7 +100,7 @@ public IFileInfo GetFileInfo(string subpath)
         public IDirectoryContents GetDirectoryContents(string subpath)
         {
             // The file name is assumed to be the remainder of the resource name.
-            if (subpath == null)
+            if (subpath == null || HasInvalidPathChars(subpath))
```

Comment:
This doesn't allow slashes in paths (other than a leading one). You could remove this check.